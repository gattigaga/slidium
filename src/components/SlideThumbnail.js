import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Container = styled.div`
  padding: 8px 30px;
  cursor: pointer;
  position: relative;
  background: ${({ isActive }) => (isActive ? "#ccc" : "#ddd")};
`;

const Content = styled.svg`
  width: 160px;
  height: 120px;
  background: white;
  box-sizing: border-box;
  border: ${({ isActive }) =>
    isActive ? "2px solid #1a85ff" : "1px solid #ccc"};
`;

const Index = styled.span`
  color: black;
  font-family: Roboto;
  font-size: 12px;
  font-weight: bold;
  position: absolute;
  top: 12px;
  left: 12px;
  user-select: none;
`;

const SlideThumbnail = ({ onClick, isActive, className, indexCaption }) => (
  <Container onClick={onClick} isActive={isActive} className={className}>
    <Index>{indexCaption}</Index>
    <Content isActive={isActive} />
  </Container>
);

SlideThumbnail.propTypes = {
  className: PropTypes.string,
  onClick: PropTypes.func,
  isActive: PropTypes.bool,
  indexCaption: PropTypes.number.isRequired
};

SlideThumbnail.defaultProps = {
  indexCaption: 0
};

export default SlideThumbnail;
