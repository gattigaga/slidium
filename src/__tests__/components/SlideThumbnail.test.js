import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import SlideThumbnail from "components/SlideThumbnail";

let wrapper, onClick;

beforeEach(() => {
  onClick = jest.fn();
  wrapper = shallow(<SlideThumbnail indexCaption={1} onClick={onClick} />);
});

describe("SlideThumbnail", () => {
  it("renders correctly", () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("renders in active", () => {
    wrapper.setProps({
      isActive: true
    });

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("renders with additional class", () => {
    wrapper.setProps({
      className: "my-class"
    });

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("calls 'onClick'", () => {
    expect(onClick.mock.calls.length).toEqual(0);
    wrapper.simulate("click");
    expect(onClick.mock.calls.length).toEqual(1);
  });
});
