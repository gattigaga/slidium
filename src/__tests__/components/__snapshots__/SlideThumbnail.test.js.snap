// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlideThumbnail renders correctly 1`] = `
<styled.div
  onClick={[Function]}
>
  <styled.span>
    1
  </styled.span>
  <styled.svg />
</styled.div>
`;

exports[`SlideThumbnail renders in active 1`] = `
<styled.div
  isActive={true}
  onClick={[Function]}
>
  <styled.span>
    1
  </styled.span>
  <styled.svg
    isActive={true}
  />
</styled.div>
`;

exports[`SlideThumbnail renders with additional class 1`] = `
<styled.div
  className="my-class"
  onClick={[Function]}
>
  <styled.span>
    1
  </styled.span>
  <styled.svg />
</styled.div>
`;
